pkg/filter/requestadaptor/requestadaptor.go

```diff
@@ -151,6 +168,62 @@ func (ra *RequestAdaptor) handle(ctx context.HTTPContext) string {
 			ctx.Request().SetHost(ra.spec.Host)
 		}
 	}
+
+	if ra.spec.Compress != "" {
+		res := ra.processCompress(ctx)
+		if res != "" {
+			return res
+		}
+	}
+
+	if ra.spec.Decompress != "" {
+		res := ra.processDecompress(ctx)
+		if res != "" {
+			return res
+		}
+	}
+	return ""
+}
+
+func (ra *RequestAdaptor) processCompress(ctx context.HTTPContext) string {
+	encoding := ctx.Request().Header().Get("Content-Encoding")
+	if encoding != "" {
+		return ""
+	}
+	var buf bytes.Buffer
+	gw := gzip.NewWriter(&buf)
+	defer gw.Close()
+
+	data, err := io.ReadAll(ctx.Request().Body())
+	if err != nil {
+		logger.Errorf("read request body failed, %v", err)
+		return resultCompressFail
+	}
+	_, err = gw.Write(data)
+	if err != nil {
+		logger.Errorf("compress request body failed, %v", err)
+		return resultCompressFail
+	}
+	ctx.Request().SetBody(&buf)
+	ctx.Request().Header().Set("Content-Encoding", "gzip")
+	return ""
+}
+
+func (ra *RequestAdaptor) processDecompress(ctx context.HTTPContext) string {
+	encoding := ctx.Request().Header().Get("Content-Encoding")
+	if ra.spec.Decompress == "gzip" && encoding == "gzip" {
+		reader, err := gzip.NewReader(ctx.Request().Body())
+		if err != nil {
+			return resultDecompressFail
+		}
+		defer reader.Close()
+		data, err := io.ReadAll(reader)
+		if err != nil {
+			return resultDecompressFail
+		}
+		ctx.Request().SetBody(bytes.NewReader(data))
```

Comment:
```suggestion
		ctx.Request().SetBody(reader)
```

Reply:
"gzip" reader is a `ReadCloser` but our `ctx.Request().Body` is `Reader`. if set `ReadCloser` directly to `Reader`, Close method will not be called and resource will not be released.....